CMakeLists.txt

```diff
@@ -7,7 +7,14 @@ include(bin/dsn.cmake)
 #project(dsn VERSION 1.0 LANGUAGES C CXX)
 project(dsn C CXX)
 set(DSN_BUILD_RUNTIME TRUE)
-set(GIT_SOURCE_FILE "ext/gitsource.cmake")
+
+if(DSN_GIT_SOURCE STREQUAL "xiaomi")
```

Comment:
done, changed as:
 if((NOT DEFINED DSN_GIT_SOURCE) OR (DSN_GIT_SOURCE STREQUAL ""))
     set(DSN_GIT_SOURCE "github")
 endif()

 message(STATUS "use git source ${DSN_GIT_SOURCE}")
 set(GIT_SOURCE_FILE "ext/gitsource.${DSN_GIT_SOURCE}")

On Wed, Jun 15, 2016 at 5:26 PM, Zhenyu Guo <notifications@github.com>
wrote:

> In CMakeLists.txt
> <https://github.com/imzhenyu/rDSN/pull/403#discussion_r67128371>:
>
> > @@ -7,7 +7,14 @@ include(bin/dsn.cmake)
> >  #project(dsn VERSION 1.0 LANGUAGES C CXX)
> >  project(dsn C CXX)
> >  set(DSN_BUILD_RUNTIME TRUE)
> > -set(GIT_SOURCE_FILE "ext/gitsource.cmake")
> > +
> > +if(DSN_GIT_SOURCE STREQUAL "xiaomi")
>
> It is best to change it to something like follows so that others may also
> use this facility.
>
> if (!defined(DSN_GIT_SOURCE) OR DSN_GIT_SOURCE STREQL "")
> set src = github
> endif
>
> set(GIT_SOURCE_FILE "ext/gitsource."$(src))
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/imzhenyu/rDSN/pull/403/files/89a41d5e513a99014291d1ad76187ae04459b1ab#r67128371>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAgQPfqZzlCqWiMf6Hw3fu0V9oA_bL_aks5qL8VdgaJpZM4I2J8D>
> .
>